homeassistant/components/roborock/coordinator.py

```diff
@@ -302,13 +302,19 @@ async def _async_update_data(self) -> DeviceProp:
             # If the vacuum is currently cleaning and it has been IMAGE_CACHE_INTERVAL
             # since the last map update, you can update the map.
             new_status = self.roborock_device_info.props.status
-            if self.current_map is not None and (
-                (
-                    new_status.in_cleaning
-                    and (dt_util.utcnow() - self.maps[self.current_map].last_updated)
-                    > IMAGE_CACHE_INTERVAL
+            if (
+                self.current_map is not None
+                and self.current_map in self.maps
+                and (
+                    (
+                        new_status.in_cleaning
+                        and (
+                            dt_util.utcnow() - self.maps[self.current_map].last_updated
+                        )
+                        > IMAGE_CACHE_INTERVAL
+                    )
+                    or self.last_update_state != new_status.state_name
                 )
```

Comment:
Typically prefer to do the dict lookup once by extracting a variable using the walrus:

```suggestion
                and current_map := self.maps.get(self.current_map)
                and (
                    (
                        new_status.in_cleaning
                        and (
                            dt_util.utcnow() - current_map.last_updated
                        )
                        > IMAGE_CACHE_INTERVAL
                    )
                    or self.last_update_state != new_status.state_name
                )
```